mlrun/datastore/dbfs_store.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import fsspec
+
+import mlrun.errors
+
+from .base import DataStore, FileStats
+
+# dbfs objects will be represented with the following URL: dbfs://<path>
+
+
+class DBFSStore(DataStore):
+    def __init__(self, parent, schema, name, endpoint="", secrets: dict = None):
+        super().__init__(parent, name, schema, endpoint, secrets=secrets)
+        if self.endpoint.startswith("https://"):
+            self.endpoint.replace("https://", "")
+        if self.endpoint.startswith("http://"):
+            self.endpoint.replace("http://", "")
+        self.get_filesystem(silent=False)
+
+    def get_filesystem(self, silent=True):
+        """return fsspec file system object, if supported"""
+        if self._filesystem:
+            return self._filesystem
+
+        self._filesystem = fsspec.filesystem("dbfs", **self.get_storage_options())
+        return self._filesystem
+
+    def get_storage_options(self):
+        return dict(
+            token=self._get_secret_or_env("DATABRICKS_TOKEN"), instance=self.endpoint
+        )
+
+    def path_and_system_validator(self, key: str):
+        if not self._filesystem:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Performing actions on data-item without a valid filesystem"
+            )
+        if not key.startswith("/"):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Invalid path(key attribute) - path must start with '/'"
+            )
+
+    def get(self, key: str, size=None, offset=0) -> bytes:
+        self.path_and_system_validator(key)
+        size_less_then_one_exception = "negative or zero size argument 1 is invalid."
+        if offset:
+            if size is None:
+                size = 1000000  # The maximum number of allowed bytes to read is 1MB.
```

Comment:
magic number alert! perhaps move to datastore/helpers.py where there you will have sizes class or so
I see that v3io.py on datastore also have constant for `ONE_MB` :)